src/server/context.ts

```diff
@@ -192,17 +194,27 @@ export class ServerContext {
         includeDirs: false,
         followSymlinks: false,
       });
+      const encoder = new TextEncoder();
       for await (const entry of entires) {
```

Comment:
Unrelated: we should parallelise this with `Promise.all` or `pooledMap` in a follow up